src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py

```diff
@@ -3194,8 +3194,114 @@ def test_sql_managed_db_short_retention(self, resource_group, resource_group_loc
+    @record_only()
+    def test_sql_managed_db_long_term_retention(
+            self):
+
+        self.kwargs.update({
+            'rg': 'clitestxj6awmetud',
+            'loc': 'westus',
+            'managed_instance_name': 'ayang-ltr-mi',
+            'database_name': 'test-1',
+            'weekly_retention': 'P1W',
+            'monthly_retention': 'P1M',
+            'yearly_retention': 'P2M',
+            'week_of_year': 12
+        })
+
+        # test update long term retention on live database
+        self.cmd(
+            'sql midb ltr-policy set -g {rg} --mi {managed_instance_name} -n {database_name} --weekly-retention {weekly_retention} --monthly-retention {monthly_retention} --yearly-retention {yearly_retention} --week-of-year {week_of_year}',
+            checks=[
+                self.check('resourceGroup', '{rg}'),
+                self.check('weeklyRetention', '{weekly_retention}'),
+                self.check('monthlyRetention', '{monthly_retention}'),
+                self.check('yearlyRetention', '{yearly_retention}')])
+
+        # test get long term retention policy on live database
+        self.cmd(
+            'sql midb ltr-policy show -g {rg} --mi {managed_instance_name} -n {database_name}',
+            checks=[
+                self.check('resourceGroup', '{rg}'),
+                self.check('weeklyRetention', '{weekly_retention}'),
+                self.check('monthlyRetention', '{monthly_retention}'),
+                self.check('yearlyRetention', '{yearly_retention}')])
+
+        # test list long term retention backups for location
+        # with resource group
+        self.cmd(
+            'sql midb ltr-backup list -l {loc} -g {rg}',
+            checks=[
+                self.greater_than('length(@)', 0)])
+
+        # without resource group
+        self.cmd(
+            'sql midb ltr-backup list -l {loc}',
+            checks=[
+                self.greater_than('length(@)', 0)])
+
+        # test list long term retention backups for instance
+        # with resource group
+        self.cmd(
+            'sql midb ltr-backup list -l {loc} --mi {managed_instance_name} -g {rg}',
+            checks=[
+                self.greater_than('length(@)', 0)])
+
+        # without resource group
+        self.cmd(
+            'sql midb ltr-backup list -l {loc} --mi {managed_instance_name}',
+            checks=[
+                self.greater_than('length(@)', 0)])
+
+        # test list long term retention backups for database
+        # with resource group
+        self.cmd(
+            'sql midb ltr-backup list -l {loc} --mi {managed_instance_name} -d {database_name} -g {rg}',
+            checks=[
+                self.greater_than('length(@)', 0)])
+
+        # without resource group
+        self.cmd(
+            'sql midb ltr-backup list -l {loc} --mi {managed_instance_name} -d {database_name}',
+            checks=[
+                self.greater_than('length(@)', 0)])
```

Comment:
Please change the check values to a specific number using something like `self.check('length(@)', 1)`

Reply:
The exact number of backups in this situation can change quickly (since backups can be added/deleted - even this test does a delete at the very end), but in general should be nonzero.

Reply:
the check is used to check the output for this command. it doesn't matter for changing later.

Reply:
changed to specific lengths